src/components/HoverTooltip.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { PropsWithChildren, ReactElement, ReactNode, useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+
+type HoverTooltipProps = {
+  /** Content to show in the Tooltip box. */
+  tooltipContent?: ReactNode;
+  /** If disabled is true, tooltip will be hidden. */
+  disabled?: boolean;
+  /** Offset of the tooltip box relative to the mouse, in pixels. By default, set to 15 pixels in x and y. */
+  offsetPx?: [number, number];
+};
+
+const defaultProps: Partial<HoverTooltipProps> = {
+  tooltipContent: <p>Tooltip</p>,
+  disabled: false,
+  offsetPx: [15, 15],
+};
+
+// Styling for the tooltip
+const TooltipDiv = styled.div`
+  position: fixed;
+
+  border-radius: var(--radius-control-small);
+  border: 1px solid var(--color-dividers);
+  background-color: var(--color-background);
+  padding: 6px 8px;
+
+  transition: opacity 300ms;
+  z-index: 1;
+`;
+
+/**
+ * Adds a tooltip region around the provided child. A tooltip will appear next to the mouse when
+ * the mouse enters the area, and disappear on exit.
+ */
```

Comment:
kinda surprised there isn't already a component for this...

Reply:
The Antd Tooltip is a little stricter, as it doesn't follow the mouse and shows in a set of fixed locations.